### PR TITLE
[16] [Integration] As a logged in user, I can go to survey detail screen and see survey detail

### LINKED ISSUE #16: [Integration] As a logged in user, I can go to survey detail screen and see survey detail
## Why

- Allow the user to go to survey detail screen
- After survey detail screen UI is ready and fetch survey detail then show the actual data
 
## Acceptance Criteria

- When circle button pressed go to survey detail screen
- Each survey detail screen must be based on `survey id`
- Display survey detail screen with actual data

### PR DESCRIPTION
**Pull request by chiiida**
https://github.com/llleyelll/ic-surveys-ios/issues/16

## What happened 👀

- Integrate Survey Detail Screen to show data from API
 
## Insight 📝

- Add unit tests
- Update Home Screen unit test

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: SurveyApp/Models/Survey/Survey.swift [hunk 1]
```diff
@@ -17,3 +19,11 @@ struct Survey: Codable, Comparable {
       return lhs.id < rhs.id
     }
 }
+
+extension Survey {
+
+    var largeImageURL: URL? {
+        URL(string: coverImageUrl + "l")
+
```
Comment (blyscuit) [inline SurveyApp/Models/Survey/Survey.swift]: extra line break here.

File: SurveyApp/Modules/Home/Cell/SurveyCollectionCell.swift [hunk 2]
```diff
@@ -33,8 +33,7 @@ final class SurveyCollectionCell: UICollectionViewCell {
     func configure(with viewModel: SurveyCollectionCellViewModel) {
         titleLabel.text = viewModel.title
         descriptionLabel.text = viewModel.description
-        guard let imageURL: URL = URL(string: viewModel.coverImageUrl) else { return }
-        imageView.af.setImage(withURL: imageURL)
+        imageView.af.setImage(withURL: viewModel.coverImageUrl!)
```
Comment (blyscuit) [inline SurveyApp/Modules/Home/Cell/SurveyCollectionCell.swift]: please don't use force unwrap.
```suggestion
        if let coverImageUrl = viewModel.coverImageUrl {
        imageView.af.setImage(withURL: coverImageUrl)
        }
```

File: SurveyAppTests/Tests/Modules/SurveyDetail/SurveyDetailPresenterSpec.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+                    presenter.survey = surveys[0]
+                    presenter.viewDidLoad()
+                }
+
+                it("triggers view to configure view") {
+                    expect(view.configureWithCalled) == true
+                }
+                
+                it("view should receive survey correctly") {
+                    expect(view.configureWithReceivedSurvey?.title) == "Scarlett Bangkok"
```
Comment (blyscuit) [inline SurveyAppTests/Tests/Modules/SurveyDetail/SurveyDetailPresenterSpec.swift]: I think you can check this with a variable 
```
expect(view.configureWithReceivedSurvey?.title) == surveys[0].title
```

**Review by blyscuit — APPROVED** [review]

**Pull request merged**